src/transports/balance.ts

```diff
@@ -0,0 +1,232 @@
+
+export type BalanceTransportGenerics = TransportGenerics & {
+  /**
+   * Type details about incoming Adapter requests to this Transport
+   */
+  Parameters: BalanceTransportInputParametersDefinition
+  /**
+   * Type details about outbound responses from this Transport
+   */
+  Response: BalanceResponse
+  /**
+   * Type details for any provider specific interfaces.
+   */
+  Provider: {
+    /**
+     * Structure of the body of the request that will be sent to the data provider.
+     */
+    RequestBody: unknown
+
+    /**
+     * Structure for the body of the response coming from the data provider.
+     */
+    ResponseBody: unknown
+  }
+}
+
+export interface BalanceAddress {
+  chain: string
+  address: string
+  coin: string
+}
+
+/**
+ * Config object that is provided to the BalanceTransport constructor.
+ */
+export interface BalanceTransportConfig<T extends BalanceTransportGenerics> {
+  /**
+   * This method is called for every address in 'addresses' input data.
+   * It takes [BalanceAddress], confirmations and adapter settings
+   * and should return single [[BalanceRequestConfig]]
+   * @param address - address object of type [BalanceAddress]
+   * @param confirmations - number of confirmations to get the balance
+   * @param adapterSettings - the config for this Adapter
+   * @returns [BalanceRequestConfig]
+   */
+  getBalance: (
+    balanceAddress: BalanceAddress,
+    confirmations: number,
+    adapterSettings: T['Settings'],
+  ) => BalanceRequestConfig<T>
+}
+
+/**
+ * Structure containing the association between input address and a provider request.
+ */
+export interface BalanceRequestConfig<T extends BalanceTransportGenerics> {
+  /** The request that will be sent to the data provider to fetch the balance of an address */
+  request: AxiosRequestConfig<T['Provider']['RequestBody']>
+  /** Function that will be called when request for an address is executed. It receives the response from the data provider
+   * and adapter settings and should return [Balance] object*/
+  onResponse: (
+    res: AxiosResponse<T['Provider']['ResponseBody']>,
+    adapterSettings: T['Settings'],
+  ) => Balance
+}
+
+interface InputData {
+  addresses: BalanceAddress[]
+  confirmations: number
+}
+
+/**
+ * Transport implementation that is meant to be used with [[BalanceEndpoint]]. It takes incoming request that
+ * contains 'addresses' array as input data. Then each address object from that array is passed through `getBalance`
+ * function, that returns AxiosRequestConfig and `onResponse` function. The Data Provider response is passed through the
+ * `onResponse` function which should return [[Balance]] object. Once the balances of all addresses are fetched and processed,
+ * the transport will cache it and return as EA response.
+ * @typeParam T - all types related to the [[Transport]]
+ */
+export class BalanceTransport<T extends BalanceTransportGenerics> implements Transport<T> {
```

Comment:
I think I agree with your point about this not being super necessary, it might be easier to understand the adapters if they just implement a custom transport. To make them shorter they could extend some of the abstract classes like SubscriptionTransport if necessary